xlators/performance/io-cache/src/page.c

```diff
@@ -797,6 +797,9 @@ ioc_frame_unwind(call_frame_t *frame)
         goto unwind;
     }
 
+    /* gnfs requires op_errno to determine is_eof */
+    op_errno = local->op_errno;
+
     if (local->op_ret < 0) {
         op_ret = local->op_ret;
         op_errno = local->op_errno;
```

Comment:
now this line is not needed?

Reply:
Are you asking me to show the test case where removing this line results in the IO Error? I can re-run that test if you like. It is in notes.txt

Let me know if I misunderstood the comment.

Reply:
Or put differently, there were two problem cases. One was sharding on its own, and one was when the NFS io cache was enabled. I am not very familiar with the code and will do whatever you want.

Reply:
All I am commenting - and it's a nit, really, is that you have redundant lines now. As you've just assigned op_errno in line 801, line 805 can be removed. It's a nit, really.

Reply:
Ok I see now what you mean. I was taking too literal the instructions earlier. Sure I can adjust that.

Reply:
retesting.... (even though it seems obvious now that you pointed it out)

Reply:
Looking good. Wrangling the patch for git from the package build I tested with. Will push it shortly and mark this resolved.